### PR TITLE
Upgrade to PHPUnit 5.7

### PR DESCRIPTION
**Pull request by hertz1**
This is required in order to run tests with Monolog 2. The changes are described in each commit.

If this PR is accepted, I will open a new one for Monolog 2 support.

### CODE REVIEW

**Comment by rantonmattei** [comment]
Ping @mymac80 @bhavinrshah or @andymc for reviews and merging requests. I am no longer the maintainer... Folks at [The Orchard  ](https://github.com/theorchard) have review and merging rights.

File: tests/Config/Loader/ClassLoader/HandlerLoaderTest.php [hunk 1]
```diff
@@ -244,8 +244,15 @@ public function testHandlerForProcessor()
     {
         $options = array();
 
-        $mockProcessor1 = '123';
-        $mockProcessor2 = '456';
+        $mockProcessor1 = function($record) {
```
Comment (bhavinrshah) [inline tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:247]: I could not get error related with this test cases. Which PHP version are you using to execute test cases?

Comment (hertz1) [reply tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:247]: I used PHP ^7.2, which is the minimum version required by **Monolog 2**, but you must update to **PHPUnit 5.7** and **Monolog 2** in you **composer.json** to be able to reproduce this.

Comment (bhavinrshah) [reply tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:247]: I used PHP ^7.4.20 and getting some different errors:

There were 3 errors:
1) Cascade\Tests\Config\Loader\ClassLoader\HandlerLoaderTest::testHandlers with data set #0 ('*', 'formatter', Monolog\Formatter\LineFormatter Object (...), 'setFormatter')
Function ReflectionType::__toString() is deprecated

/var/www/html_staging/monolog-cascade-new/tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:169
/var/www/html_staging/monolog-cascade-new/tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:237

2) Cascade\Tests\Config\Loader\ClassLoader\Resolver\ExtraOptionsResolverTest::testResolveWithCustomOptionHandler
Function ReflectionType::__toString() is deprecated

/var/www/html_staging/monolog-cascade-new/tests/Config/Loader/ClassLoader/Resolver/ExtraOptionsResolverTest.php:132

3) Cascade\Tests\ConfigTest::testConfigure
Function ReflectionType::__toString() is deprecated

/var/www/html_staging/monolog-cascade-new/tests/ConfigTest.php:66

Comment (hertz1) [reply tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:247]: I've managed to reproduced these errors. By default, the `php:7.4-cli` docker image doesn't have deprecation errors enabled.

Apparently, this deprecation notice was [introduced in PHP 7.4](https://www.php.net/manual/en/reflectiontype.tostring.php#125211).

I've tracked down the source of the problem to be [this line in phpunit-mock-objects](https://github.com/sebastianbergmann/phpunit-mock-objects/blob/0dfddc236629eb7ead4df2dfa3d79d548e011a3b/src/Generator.php#L1153).

PHPUnit [removed the dependency](https://github.com/sebastianbergmann/phpunit/commit/35dcfdc6d1196e32629d5c76d3ee8b0a0093065e#diff-d2ab9925cad7eac58e0ff4cc0d251a937ecf49e4b6bf57f8b95aab76648a9d34) on this repository starting from version 7.2.0, so I think this should be the minimum required version, although a more recent version of PHPUnit would be better.

I'll close this PR and open a new one for more recent PHPUnit support.

Comment (bhavinrshah) [reply tests/Config/Loader/ClassLoader/HandlerLoaderTest.php:247]: Thanks @hertz1

**Pull request closed without merging**